core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -4559,6 +4559,46 @@ private void checkNullOperand(SqlTester tester, String op) {
         });
   }
 
+  @Test public void testLambda() {
+    tester.checkType("(a)->2+2*a", "LAMBDA NOT NULL");
+    tester.checkType("(a,b)->a+b", "LAMBDA NOT NULL");
+    tester.checkType("(a,b,c)->2*a+b+c", "LAMBDA NOT NULL");
+  }
+
+  @Test public void testMapFilterFunc() {
+    final SqlTester tester = tester(SqlLibrary.MYSQL);
```

Comment:
I think we should also add some negative tests. For instance, `tester.checkType("(b)->2+2*a")`.

Reply:
Added few more test cases